### PR TITLE
Add Front-End Unit Tests

### PR DESCRIPTION
**Pull request by jwyatt1999**
This PR adds unit tests for the front-end of the website, and also includes a few bug fixes that had to be made to get the unit tests to pass.

- [x] Ran unit tests, 6/6 passed with runtime of 41.898s

Bug fixes:
- Fixed crash when user has no posted scores, since round() causes a crash when NoneType is input
- Fixed annoying issue where averages that were integers were shown with 1 decimal places but averages that were floats were shown with 2 decimal places
- Fixed issue where default value of leaderboard page button was 01-05-2022, now it defaults to the current date
- Fixed issue where value of leaderboard page button was being reset to 01-05-20222 after each click, now it stays on the value which the user selected last
- Fixed issue where under some rare circumstances some pieces could have their type swapped to grant a coin even though the shape wasn't changing (the pieces should be locked into granting / not granting a coin in these cases)

### CODE REVIEW

**Comment by jwyatt1999** [comment]
Hey @Bsubs , this PR is now ready for review :)

I would be happy to demo the user tests to you tomorrow or on Saturday, whatever works for you

File: seleniumtests.py [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+        #Check that we are logged in
+        self.assertEqual(self.driver.current_url, "http://127.0.0.1:5000/")
+        #Navigate to leaderboard page
+        self.driver.get('http://127.0.0.1:5000/leaderboard')
+        self.driver.implicitly_wait(5)
+        #Check daily scores and this user's highest daily attempt
+        daily_highestScore = self.driver.find_elements_by_name('dailyScore')[0].text
+        daily_scoreCount = len(self.driver.find_elements_by_name('dailyScore'))
+        daily_myHighestAttempt = self.driver.find_element_by_id('leaderboard_dailyHighscore').text
+        self.assertEqual(daily_highestScore, "test3 scored 40 points on 2022-05-18")
```
Comment (Bsubs) [inline seleniumtests.py]: The hardcoded date will fail tests as scorecards will increment dates based on server time

File: seleniumtests.py [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+        self.assertEqual(daily_scoreCount, 9)
+        self.assertEqual(daily_myHighestAttempt, "test1 scored 15 points on 2022-05-18")
```
Comment (Bsubs) [inline seleniumtests.py]: Same as above.

**Comment by Bsubs** [comment]
Besides the two small changes everything looks good :)

File: seleniumtests.py (hunk 1 above)
Comment (jwyatt1999) [reply seleniumtests.py]: Good catch, I didn't think about that. I'll just compare the part that says "test3 scored 40 points on"

**Comment by Bsubs** [comment]
Nice :) I'll merge the PR in now.

**Pull request merged**